File: cinema_app_mobile/web/cinema_app_backend/models/Reservation.js
const mongoose = require('mongoose');
let reservationSchema = mongoose.Schema(
    {
      idUser:{type: mongoose.Schema.Types.ObjectId,ref: "Users"},
      idMovie:{type: mongoose.Schema.Types.ObjectId,ref: "Movie"},
      nbr:{type:Number ,required:true}
})




let Reservation = mongoose.model('Reservation',reservationSchema)

module.exports = Reservation

File: cinema_app_mobile/lib/models/reservation.dart
class Reservation {
  String id;
  dynamic movie;
  int nbr;

  Reservation({required this.id, required this.movie, required this.nbr});

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
        id: json['_id'], movie: json['idMovie'], nbr: json['nbr']);
  }
}
